llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -1464,9 +1464,11 @@ void VFABI::getVectorVariantNames(
 
   for (const auto &S : SetVector<StringRef>(ListAttr.begin(), ListAttr.end())) {
 #ifndef NDEBUG
+    SmallVector<Type *> OpTys;
+    for (Value *Op : CI.operands())
+      OpTys.push_back(Op->getType());
     LLVM_DEBUG(dbgs() << "VFABI: adding mapping '" << S << "'\n");
-    std::optional<VFInfo> Info =
-        VFABI::tryDemangleForVFABI(S, *(CI.getModule()));
+    std::optional<VFInfo> Info = VFABI::tryDemangleForVFABI(S, OpTys);
```

Comment:
You previously identified the operands might not be enough so perhaps just pass in the call instruction itself rather than spend time pulling the operands into a separate vector.